### PR TITLE
`System.CommandLine` upgrade fixes

### PR DESCRIPTION
**Pull request by yuchong-pan**
1. Updated `VERSION` parameter name of root command. However, the parameter `VERSIONS` of `--all-but` does not change.

    Before upgrade:
    ```
    dotnet-core-uninstall [options] <VERSIONS> [command]
    ```
    After:
    ```
    dotnet-core-uninstall [options] [<VERSION>...] [command]
    ```
2. Hid the `--version` option (the 0.3.0 version of `System.CommandLine` removes `isHidden` from the constructor parameters).
3. Moved `BundleTypePrintInfo` from `Shared.Commands` to `Shared.Configs`. Renamed `Windows.ListCommandExec` and `MacOs.ListCommandExec` to `SupportedBundleTypeConfigs`.
4. Removed `--aspnet-runtime` and `--hosting-bundle` from the macOS version.
5. Added separate option descriptions for the `list` command to align with #54.

### CODE REVIEW

File: test/dotnet-core-uninstall.Tests/Shared/Configs/CommandLineConfigsTests.cs [hunk 1]
```diff
@@ -384,7 +434,32 @@ internal void TestGetUninstallMainOptionVersionBeforeOptionException(string opti
...
+        [InlineData("--runtime", BundleType.Runtime)]
+        [InlineData("--sdk --runtime", BundleType.Sdk | BundleType.Runtime)]
+        [InlineData("-v q", BundleType.Sdk | BundleType.Runtime)]
+        [InlineData("--sdk --verbosity minimal", BundleType.Sdk)]
+        [InlineData("-v normal --runtime", BundleType.Runtime)]
+        [InlineData("--sdk --verbosity diag --runtime", BundleType.Sdk | BundleType.Runtime)]
+        [InlineData("--all", BundleType.Sdk | BundleType.Runtime)]
+        [InlineData("--sdk --all-but 2.2.300 2.1.700", BundleType.Sdk)]
+        [InlineData("--runtime --all-below 3.0.1-preview-10086", BundleType.Runtime)]
+        [InlineData("--sdk --runtime --all-previews", BundleType.Sdk | BundleType.Runtime)]
```
Comment (wli3) [inline test/dotnet-core-uninstall.Tests/Shared/Configs/CommandLineConfigsTests.cs]: suggestion, add "`--aspnet-runtime` will throw in macos" tests

File: src/dotnet-core-uninstall/Shared/Configs/BundleTypePrintInfo.cs [hunk 2]
```diff
@@ -4,19 +4,21 @@
...
-namespace Microsoft.DotNet.Tools.Uninstall.Shared.Commands
+namespace Microsoft.DotNet.Tools.Uninstall.Shared.Configs
 {
     internal abstract class BundleTypePrintInfo
     {
         public abstract BundleType Type { get; }
 
         public string Header { get; }
         public Func<IList<Bundle>, GridView> GridViewGenerator { get; }
+        public string OptionName { get; }
```
Comment (wli3) [inline src/dotnet-core-uninstall/Shared/Configs/BundleTypePrintInfo.cs]: nit: maybe something like commandline option. It sounds like the "name" is optional

Comment (yuchong-pan) [reply src/dotnet-core-uninstall/Shared/Configs/BundleTypePrintInfo.cs]: I think the name `Option` has a very specific meaning throughout the entire repository; it means the command line option. Hence I still feel like `OptionName` is clear enough for this field.

File: test/dotnet-core-uninstall.Tests/Shared/Configs/CommandLineConfigsTests.cs (hunk 1 above)
Comment (yuchong-pan) [reply test/dotnet-core-uninstall.Tests/Shared/Configs/CommandLineConfigsTests.cs]: Thanks for your suggestion! It's been added.

**Pull request merged**